aeppl/censoring.py

```diff
@@ -0,0 +1,214 @@
+        if base_rv.dtype.startswith("int"):
+            logccdf = at.logaddexp(logccdf, logprob)
+
+        logprob = at.switch(
+            at.eq(value, upper_bound),
+            logccdf,
+            at.switch(at.gt(value, upper_bound), -np.inf, logprob),
+        )
+    if not (
+        isinstance(lower_bound, TensorConstant)
+        and np.all(np.isneginf(lower_bound.value))
+    ):
+        is_lower_bounded = True
+        logprob = at.switch(
+            at.eq(value, lower_bound),
+            logcdf,
+            at.switch(at.lt(value, lower_bound), -np.inf, logprob),
+        )
+
+    if is_lower_bounded and is_upper_bounded:
+        logprob = CheckParameterValue("lower_bound <= upper_bound")(
+            logprob, at.all(at.le(lower_bound, upper_bound))
+        )
+
+    return logprob
+
+
+class MeasurableRound(Elemwise):
+    """A placeholder used to specify a log-likelihood for a clipped RV sub-graph."""
+
+
+MeasurableVariable.register(MeasurableRound)
+
+
+@node_rewriter(tracks=[Elemwise])
+def find_measurable_roundings(
+    fgraph: FunctionGraph, node: Node
+) -> Optional[List[MeasurableRound]]:
+    # TODO: Canonicalize x[x>ub] = ub -> clip(x, x, ub)
+
+    rv_map_feature = getattr(fgraph, "preserve_rv_mappings", None)
+    if rv_map_feature is None:
+        return None  # pragma: no cover
+
+    if isinstance(node.op, MeasurableRound):
+        return None  # pragma: no cover
+
+    if not (
+        isinstance(node.op, Elemwise)
+        and isinstance(node.op.scalar_op, (RoundHalfToEven, Floor, Ceil))
+    ):
+        return None
+
+    (rounded_var,) = node.outputs
+    (base_var,) = node.inputs
+
+    if not (
+        base_var.owner
+        and isinstance(base_var.owner.op, MeasurableVariable)
+        and base_var not in rv_map_feature.rv_values
+        # Rounding only makes sense for continuous variables
+        and base_var.dtype.startswith("float")
+    ):
+        return None
+
+    # Make base_var unmeasurable
+    unmeasurable_base_var = assign_custom_measurable_outputs(base_var.owner)
+
+    rounded_op = MeasurableRound(node.op.scalar_op)
+    rounded_rv = rounded_op.make_node(unmeasurable_base_var).default_output()
+    rounded_rv.name = rounded_var.name
+    return [rounded_rv]
+
+
+measurable_ir_rewrites_db.register(
+    "find_measurable_roundings",
+    find_measurable_roundings,
+    0,
+    "basic",
+    "censoring",
+)
+
+
+@_logprob.register(MeasurableRound)
+def round_logprob(op, values, base_rv, **kwargs):
+    (value,) = values
+
+    if isinstance(op.scalar_op, RoundHalfToEven):
+        value = at.round(value)
+        value_upper = value + 0.5
+        value_lower = value - 0.5
+    elif isinstance(op.scalar_op, Floor):
+        value = at.floor(value)
+        value_upper = value + 1.0
+        value_lower = value
+    elif isinstance(op.scalar_op, Ceil):
+        value = at.ceil(value)
+        value_upper = value
+        value_lower = value - 1.0
```

Comment:
I am not sure about rounding the values. In theory they should be rounded already...